### DISCUSSION TITLE
Elliptic Curve Digital Signature Algorithm

### DISCUSSION THREAD

**Discussion by Opex25** [Cyfrin/foundry-full-course-cu#2077] [General]
Hello everyone, I am relatively new to the world of web3. My question is if the private key is passed through the Elliptic Curve Digital Signature Algorithm to get the public key, is there anyway to reverse engineer the process to get the private key and if so, how safe is it to show your public key

**Comment by EngrPips** [comment]
It seems like you figured this out by just following the course. Keep crushing it.

**Comment by Opex25** [comment]
Yes, thanks

**Reply by EngrPips** [reply]
Interesting.
